tasks/test.py

```diff
@@ -92,11 +121,128 @@ def install_tools(ctx):
+    Runs linters for given flavor, build tags, and modules.
+    """
+    print(f"--- Flavor {flavor.name}: vet and staticcheck (legacy)")
+    for module in modules:
+        print(f"----- Module '{module.full_path()}'")
+        if not module.condition():
+            print("----- Skipped")
+            continue
+
+        with ctx.cd(module.full_path()):
+            vet(ctx, targets=module.targets, rtloader_root=rtloader_root, build_tags=build_tags, arch=arch)
+            staticcheck(ctx, targets=module.targets, build_tags=build_tags, arch=arch)
+
+    # For now we only run golangci_lint on Unix as the Windows env needs more work
+    if sys.platform != 'win32':
+        print(f"--- Flavor {flavor.name}: golangci_lint")
+        for module in modules:
+            print(f"----- Module '{module.full_path()}'")
+            if not module.condition():
+                print("----- Skipped")
+                continue
+
+            with ctx.cd(module.full_path()):
+                golangci_lint(
+                    ctx, targets=module.targets, rtloader_root=rtloader_root, build_tags=build_tags, arch=arch
+                )
+
+
+def test_flavor(
+    ctx,
+    flavor: AgentFlavor,
+    build_tags: List[str],
+    modules: List[GoModule],
+    cmd: str,
+    env: Dict[str, str],
+    args: Dict[str, str],
+    junit_tar: str,
+    save_result_json: str,
+    test_profiler: TestProfiler,
+):
+    """
+    Runs unit tests for given flavor, build tags, and modules.
+    """
+    print(f"--- Flavor {flavor.name}: unit tests")
+
+    failed_modules = []
+    junit_files = []
+
+    args["go_build_tags"] = " ".join(build_tags + ["test"])
+
+    junit_file_flag = ""
+    junit_file = f"junit-out-{flavor.name}.xml"
+    if junit_tar:
+        junit_file_flag = "--junitfile " + junit_file
+    args["junit_file_flag"] = junit_file_flag
+
+    for module in modules:
+        print(f"----- Module '{module.full_path()}'")
+        if not module.condition():
+            print("----- Skipped")
+            continue
+
+        with ctx.cd(module.full_path()):
+            res = ctx.run(
+                cmd.format(
+                    packages=' '.join(f"{t}/..." if not t.endswith("/...") else t for t in module.targets), **args
+                ),
+                env=env,
+                out_stream=test_profiler,
+                warn=True,
+            )
+
+        if res.exited is None or res.exited > 0:
+            failed_modules.append(module.full_path())
+
+        if save_result_json:
+            with open(save_result_json, 'ab') as json_file, open(
+                os.path.join(module.full_path(), GO_TEST_RESULT_TMP_JSON), 'rb'
+            ) as module_file:
+                json_file.write(module_file.read())
+
+        junit_file_path = None
+        if junit_tar:
+            junit_file_path = os.path.join(module.full_path(), junit_file)
+            add_flavor_to_junitxml(junit_file_path, flavor)
+
+        junit_files.append(junit_file_path)
```

Comment:
Do we want to append even if `junit_tar` is `None`/`""` (and thus `junit_file_path` is `None`)? Not sure I understand this very well

Reply:
I think it ultimately doesn't matter, as if `junit_tar` is `None` / `""`, then the `junit_files` won't be used. I'll still move this under the condition, as that makes more sense.